File: problemSet/41A-Translation.py
#!/usr/bin/env python3
"""
Codeforces
41 A. Translation

@author yamaton
@date 2015-07-28
"""


def tf_to_yn(tf):
    return 'YES' if tf else 'NO'


def solve(s, t):
    return s[::-1] == t


def main():
    s = input().strip()
    t = input().strip()
    result = solve(s, t)
    print(tf_to_yn(result))


if __name__ == '__main__':
    main()

File: problemSet/41A_Translation.cpp
/*
 * Codeforces
 *
 * Problem 41 A. Translation
 *
 * author: yamaton
 * date: 2015-11-26
 * 
 */
#include <algorithm>
#include <iostream>
#include <vector>
#include <iterator>

using namespace std;


string tf_to_yn(bool tf) {
    return (tf) ? "YES" : "NO";
}


bool solve2(string s, string t) {
    return s.size() == t.size() && equal(s.begin(), s.end(), t.rbegin());
}


bool solve(string s, string t) {
    reverse(s.begin(), s.end());
    return s == t;
}


int main() {
    string s, t;
    cin >> s >> t;
    cout << tf_to_yn(solve2(s, t)) << endl;
    return 0;
}